Jenkinsfile

```diff
@@ -129,15 +141,42 @@ pipeline {
                 }
             }
         }
+
         stage('Integration') {
-            agent any
-            steps {
-                unstash 'StanSetup'
-                setupCXX()
-                runTests("src/test/integration", separateMakeStep=false)
+            parallel {
+                stage('Integration Windows') {
+                    agent { label 'windows' }
+                    steps {
+                        deleteDirWin()
+                            unstash 'StanSetup'
+                            setupCXX()
+                            bat "make -j${env.PARALLEL} test-headers"
+                            setupCXX(false)
+                            runTestsWin("src/test/integration")
```

Comment:
I think this is right. But I also had the impression a separate make step adds a bunch of time on Windows - is that true? If it adds more than a few minutes we could just change this to separateMakeStep=false to save time.

Reply:
This takes 5 minutes, check it out here: https://jenkins.mc-stan.org/blue/organizations/jenkins/Stan/detail/PR-2761/10/pipeline